pkgs/pkgs-lib/formats.nix

```
@@ -123,6 +123,55 @@ rec {
 
   };
 
+  keyValue = {
+    # Represents lists as duplicate keys
+    listsAsDuplicateKeys ? false,
+    # Alternative to listsAsDuplicateKeys, converts list to non-list
+    # listToValue :: [Atom] -> Atom
+    listToValue ? null,
+    ...
+    }@args:
+    assert !listsAsDuplicateKeys || listToValue == null;
+    {
+
+    type = with lib.types; let
+
+      singleAtom = nullOr (oneOf [
+        bool
+        int
+        float
+        str
+      ]) // {
+        description = "atom (null, bool, int, float or string)";
```

```suggestion
        description = "atom (string, bool, int, null or float)";
```

In presumable order of likelihood.

---

Would it make sense to
 - make this type overridable as a parameter?
 - make this type reusable elsewhere? (See other formats. Really not sure tbh)
 - add `path`?

---

> > add path?
> 
> maybe? we should check why it wasn't added to other formats

Perhaps because explicit is good. Implicit string coercions can be considered a mistake.

> Should I modify both?

Feel free.



> > make this type overridable as a parameter?
> 
> how would the conversion to string work?

Provided in the type, like `coercedTo` does, perhaps. We can scope this out.



> > make this type reusable elsewhere?
> 
> This is slightly different from others because it only allows top-level definitions

:+1:

---

> > > In presumable order of likelihood.

> >Should I modify both?

> Feel free.

I prefer to keep the order like this then.

That was the only remaining issue so I believe this thread is resolved